components/authz-service/storage/postgres/migration/sql/55_use_project_db_id_in_references_of_project_roles.up.sql

```diff
@@ -0,0 +1,102 @@
+BEGIN;
+-- break all references
+ALTER TABLE iam_role_projects
+    DROP CONSTRAINT iam_role_projects_project_id_fkey;
+
+ALTER TABLE iam_role_projects RENAME COLUMN project_id TO project_temp_id;
+ALTER TABLE iam_role_projects
+    ADD COLUMN project_id INTEGER REFERENCES iam_projects (db_id) ON DELETE CASCADE DEFERRABLE;
+UPDATE
+    iam_role_projects t
+SET
+    project_id = (
+        SELECT
+            db_id
+        FROM
+            iam_projects
+        WHERE
+            id = t.project_temp_id);
+ALTER TABLE iam_role_projects
+    DROP COLUMN project_temp_id;
+
+-- returns NULL if there's no associated projects; the call can decide
+-- (by coalesce'ing) if they want that to be {} or {(unassigned)}.
+CREATE OR REPLACE FUNCTION
+  role_projects(_role_id TEXT, OUT _project_ids TEXT[])
+  RETURNS TEXT[] AS $$
+
+  DECLARE
+  role_db_id INTEGER;
+  BEGIN
+      SELECT db_id INTO STRICT role_db_id
+      FROM iam_roles AS r
+      WHERE r.id=_role_id;
+      IF FOUND THEN
```

Comment:
I am torn... a successful find sets `FOUND`, yes, but an unsuccessful find throws an exception. So you do not really need the `IF FOUND...`; it will work exactly the same without it.

However... on the one hand, it is more clear to have `IF FOUND...` because you know that's what the author really meant. But on the other hand, it makes one think (incorrectly) that a 'not found' case just falls through.
🤔 Not sure what is "better". What do you think about keeping the `IF FOUND` and just above the `END IF` add a comment `-- ELSE throw SQLException` ?

Reply:
I've been thinking more about this one, too. At some point, I wanted it to return `{(unassigned)}`, but then I've reconsidered. And I think you're right, if we don't do that, we don't need the `IF FOUND` conditional. 👍